### PR TITLE
Add Lock's PdoStore documentation

### PR DESCRIPTION
**Pull request by jderusse**
Documenting https://github.com/symfony/symfony/pull/27456

### CODE REVIEW

File: components/lock.rst [hunk 1]
```diff
@@ -195,7 +196,7 @@ PHP process is terminated::
 
     Beware that some file systems (such as some types of NFS) do not support
     locking. In those cases, it's better to use a directory on a local disk
-    drive or a remote store based on Redis or Memcached.
+    drive or a remote store based on Pdo, Redis or Memcached.
```
Comment (OskarStark) [inline components/lock.rst]: Pdo-> PDO

File: components/lock.rst [hunk 2]
```diff
@@ -431,6 +475,30 @@ method uses the Memcached's ``flush()`` method which purges and removes everythi
...
 
+PdoStore
+~~~~~~~~~~
+
+Pdo stores rely on the `ACID`_ properties of the SQL engine.
+
+.. caution::
+
+    In a cluster configured with multiple master, ensure writes are
+    synchronously propaged to every nodes, or always use the same node.
```
Comment (OskarStark) [inline components/lock.rst]: propagated?

File: components/lock.rst [hunk 3]
```diff
@@ -431,6 +475,30 @@ method uses the Memcached's ``flush()`` method which purges and removes everythi
...
+
+.. caution::
+
+    Some SQL engine like MySQL allows to disable unique constraint check.
+    Ensure that this is not the case ``SET unique_checks=1;``.
+
+In order to purge old lock, this store use a current datetime to define a
+expiration date reference. This mechanism relies on all client and server nodes
+to have synchronized clock.
```
Comment (OskarStark) [inline components/lock.rst]: I read the code PR, is a synchronized clock still needed?

Comment (jderusse) [reply components/lock.rst]: Good catch! indeed a previous version uses the client's clock. But it has been replaced by SGDB's  time. 

I removed the `client` from the sentence, as this is still the case for multi-master nodes

File: components/lock.rst [hunk 4]
```diff
@@ -217,6 +218,49 @@ support blocking, and expects a TTL to avoid stalled locks::
...
+    The PdoStore was introduced Symfony 4.2.
+
+
+The PdoStore saves locks in an SQL database, it requires a `PDO`_,
+`Doctrine DBAL Connection`_, or `Data Source Name (DSN)`_. This store does not
+support blocking, and expects a TTL to avoid stalled locks::
+
+    use Symfony\Component\Lock\Store\PdoStore;
+
+
```
Comment (fabpot) [inline components/lock.rst]: extra blank line

File: components/lock.rst [hunk 5]
```diff
@@ -431,6 +475,30 @@ method uses the Memcached's ``flush()`` method which purges and removes everythi
     The method ``flush()`` must not be called, or locks should be stored in a
     dedicated Memcached service away from Cache.
 
+PdoStore
+~~~~~~~~~~
+
+The PdoStore rely on the `ACID`_ properties of the SQL engine.
```
Comment (fabpot) [inline components/lock.rst]: relies

File: components/lock.rst [hunk 6]
```diff
@@ -431,6 +475,30 @@ method uses the Memcached's ``flush()`` method which purges and removes everythi
...
+
+.. caution::
+
+    In a cluster configured with multiple master, ensure writes are
```
Comment (fabpot) [inline components/lock.rst]: multiple masters

File: components/lock.rst [hunk 7]
```diff
@@ -431,6 +475,30 @@ method uses the Memcached's ``flush()`` method which purges and removes everythi
...
+    synchronously propagated to every nodes, or always use the same node.
+
+.. caution::
+
+    Some SQL engine like MySQL allows to disable unique constraint check.
```
Comment (fabpot) [inline components/lock.rst]: SQL engines ... allow ... disable the unique ...

File: components/lock.rst [hunk 8]
```diff
@@ -431,6 +475,30 @@ method uses the Memcached's ``flush()`` method which purges and removes everythi
...
+    Ensure that this is not the case ``SET unique_checks=1;``.
+
+In order to purge old lock, this store use a current datetime to define a
```
Comment (fabpot) [inline components/lock.rst]: old locks, this store uses

Comment (fabpot) [reply components/lock.rst]: an expiration

File: components/lock.rst [hunk 9]
```diff
@@ -431,6 +475,30 @@ method uses the Memcached's ``flush()`` method which purges and removes everythi
...
+expiration date reference. This mechanism relies on all server nodes to
+have synchronized clock.
```
Comment (fabpot) [inline components/lock.rst]: clocks

File: components/lock.rst [hunk 10]
```diff
@@ -431,6 +475,30 @@ method uses the Memcached's ``flush()`` method which purges and removes everythi
...
+
+.. caution::
+
+    To ensure locks don't expire prematurely; the ttl's should be set with
```
Comment (fabpot) [inline components/lock.rst]: TTLs

File: components/lock.rst [hunk 11]
```diff
@@ -290,11 +334,11 @@ the component is used in the following way.
...
 ~~~~~~~~~~~~~
 
-Remote stores (:ref:`MemcachedStore <lock-store-memcached>` and
-:ref:`RedisStore <lock-store-redis>`) use an unique token to recognize the true
-owner of the lock. This token is stored in the
-:class:`Symfony\\Component\\Lock\\Key` object and is used internally by the
-``Lock``, therefore this key must not be shared between processes (session,
+Remote stores (:ref:`MemcachedStore <lock-store-memcached>`,
+:ref:`PdoStore <lock-store-pdo>` and :ref:`RedisStore <lock-store-redis>`) use
+an unique token to recognize the true owner of the lock. This token is stored
```
Comment (fabpot) [inline components/lock.rst]: a unique

**Review by fabpot — CHANGES_REQUESTED** [review]

File: components/lock.rst [hunk 12]
```diff
@@ -431,6 +474,30 @@ method uses the Memcached's ``flush()`` method which purges and removes everythi
...
     dedicated Memcached service away from Cache.
 
+PdoStore
+~~~~~~~~~~
+
+The PdoStore relies on the `ACID`_ properties of the SQL engine.
+
+.. caution::
+
+    In a cluster configured with multiple masters, ensure writes are
```
Comment (jderusse) [inline components/lock.rst]: Is it correct to use the word `master` (in the context master/slave database)?

ping @lsmith77 or @javiereguiluz

Comment (fabpot) [reply components/lock.rst]: master/slave is a bad terminology (see https://twitter.com/dhh/status/1032268675674296320)

**Comment by javiereguiluz** [comment]
Thanks for contributing these docs Jérémy! They are merged now in time to prepare for Symfony 4.2 update. Thanks!

**Pull request closed without merging**